src/Microsoft.AspNet.Mvc.DataAnnotations/DataAnnotationsModelValidator.cs

```diff
@@ -22,13 +30,39 @@ public DataAnnotationsModelValidator(ValidationAttribute attribute, IStringLocal
 
             Attribute = attribute;
             _stringLocalizer = stringLocalizer;
+            _validationAttributeAdapterProvider = validationAttributeAdapterProvider;
         }
 
+        /// <summary>
+        /// The attribute being validated against.
+        /// </summary>
         public ValidationAttribute Attribute { get; }
 
+        /// <summary>
+        /// Validates the context against the <see cref="ValidationAttribute"/>.
+        /// </summary>
+        /// <param name="validationContext">The context being validated.</param>
+        /// <returns>An enumerable of the validation results.</returns>
         public IEnumerable<ModelValidationResult> Validate(ModelValidationContext validationContext)
         {
-            var metadata = validationContext.Metadata;
+            if (validationContext == null)
+            {
+                throw new ArgumentNullException(nameof(validationContext));
+            }
+            if (validationContext.ModelMetadata == null)
+            {
+                throw new ArgumentException(Resources.FormatPropertyOfTypeCannotBeNull(
+                    nameof(validationContext.ModelMetadata),
+                    nameof(validationContext)));
+            }
+            if (validationContext.MetadataProvider == null)
+            {
+                throw new ArgumentException(Resources.FormatPropertyOfTypeCannotBeNull(
+                    nameof(validationContext.MetadataProvider),
+                    nameof(validationContext)));
```

Comment:
No, full statement s/b:
``` c#
                throw new ArgumentException(
                    Resources.FormatPropertyOfTypeCannotBeNull(
                        nameof(validationContext.ModelMetadata),
                        typeof(ModelValidationContext)),
                    nameof(validationContext));
```
Then the final error will read something like "The 'ModelMetadata' property of 'Microsoft.AspNet...ModelValidationContext' must not be null.\nParameter: validationContext."